lib/src/native/realm_core.dart

```diff
@@ -51,7 +51,7 @@ late RealmLibrary _realmLib;
 
 final _RealmCore realmCore = _RealmCore();
 
-class _RealmCore {
+class _RealmCore implements RealmCoreScheduler {
```

Comment:
I need `RealmCoreScheduler` in order to define this constructor ` Scheduler.init(RealmCoreScheduler realmCore)` in `scheduler.dart`, since `_RealmCore` is not accessible outside the file `realm_core.dart`.

Reply:
I don't think we need to do this. We are introducing an interface that only one class will be ever implementing. Also RealmCore is not part of the SDK as a hierarchy class. Its purpose is to initialize and abstract Core native access. Lets discuss this

Reply:
I returned this class back `RealmCoreScheduler` and the Scheduler initialization on `RealmCore`, in order to fix a bug for the following case:
If the users call `Realm(config)` only they should see the default messages printed. The event was not emitted unless they call `Realm.logger` getter.